core/src/apps/bitcoin/sign_tx/bitcoin.py

```diff
@@ -122,27 +139,26 @@ def __init__(
         # stable device tests.
         self.orig_txs: list[OriginalTxInfo] = []
 
-        # The digests of the external inputs streamed for approval in Step 1. These are used
-        # to ensure that the inputs streamed for verification in Step 3 are the same as
+        # The digest of the presigned external inputs streamed for approval in Step 1. This is
+        # used to ensure that the inputs streamed for verification in Step 3 are the same as
```

Comment:
one thing I still don't understand is why we can skip over inputs that are not presigned.

Reply:
There are three types of external inputs:
- with ownership proof - These used to be checked in Step 3, but are now checked in Step 1, see `process_external_input()`.
- presigned - These are still checked in Step 3.
- unverified - There is nothing to check for these. Only allowed when safety checks are disabled. Requires user confirmation if present, see `approvers.py`.